src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/SearchParameterExpressionParser.cs

```diff
@@ -202,17 +208,53 @@ public SearchParameterExpressionParser(IReferenceSearchValueParser referenceSear
                 Expression[] expressions = parts.Select(part =>
                 {
                     ISearchValue searchValue = parser(part);
+                    searchValue = ApplyTargetTypeModifier(searchValue);
 
                     return helper.Build(
                         searchParameter.Name,
                         modifier,
+                        targetTypeModifier,
                         comparator,
                         componentIndex,
                         searchValue);
                 }).ToArray();
 
                 return Expression.Or(expressions);
             }
+
+            ISearchValue ApplyTargetTypeModifier(ISearchValue source)
+            {
+                var referenceSearchValue = source as ReferenceSearchValue;
+                if (referenceSearchValue == null || string.IsNullOrEmpty(targetTypeModifier))
```

Comment:
>targetTypeModifier [](start = 73, length = 18)

Took me a while to figure out this is local function which doing variable capturing from another upper function.
It's reads a bit weird, to pass one value as parameter, but capture another one by context.
Maybe for readability reasons, can we pass  `targetTypeModifier` as parameter as well?

Reply:
Is there any guidance on this type of thing in the .net docs? Else, I would say this is a nit, it is a design feature of local functions